src/network/plugins/base/base-network-plugin.ts

```diff
@@ -61,6 +67,18 @@ export default class BaseNetworkPlugin extends Events {
       this.defaultLogger(`peer not found local %p`, peerId)
       let routingPeer = await this.network.libp2p.peerRouting.findPeer(peerId);
 
+      //set timestamp on newly found peer
+      const timestamp = Date.now();
+      const uint8Array = uint8ArrayFromString(`${timestamp}`);
+      try{
+        this.network.libp2p.peerStore.patch(peerId, {
```

Comment:
Why patching here?
Can't we patch addresses and timestamp at the same time?

Reply:
and what if routingPeer is null?

Reply:
> and what if routingPeer is null?

routingPeer won't be null in any case Because if findPeer fails it goes into catch and it wont return null object.
We can still check if its null or not but its redundant. Shall we check if its null?

Reply:
> > Why patching here? Can't we patch addresses and timestamp at the same time?
> 
> The patching of address has a different condition: `if(peer && !peer.addresses.length)`. I don't want to change this condition. If we change it to `if(peer)`, we can patch both the address and metadata with a single patch. Should we do this?

The code is buggy now. When a node changes its address, peerStore does not update the address and the first address remains in peerStore forever.

When you work on a code, your code will be the final version and need to make sure that all parts(including the old parts) are working fine. So if you need to change a condition/function, you can do it or if you do not know the logic of old parts, we can discuss.